starlette_context/middleware/context_middleware.py

```diff
@@ -43,15 +49,22 @@ async def set_context(self, request: Request) -> dict:
     async def dispatch(
         self, request: Request, call_next: RequestResponseEndpoint
     ) -> Response:
-        _starlette_context_token: Token = _request_scope_context_storage.set(
-            await self.set_context(request)
-        )
+        try:
+            context = await self.set_context(request)
+            token: Token = _request_scope_context_storage.set(context)
+        except ValueError as e:
```

Comment:
I'm still not sure if it's a good idea to handle it like that but if we were to proceed it can't be `ValueError` as something else might raise it. Only errors we know we want to handle here should be handled, otherwise 500 is correct. On master you can find `errors.py`. Maybe it's time to introduce `StarletteContextError` base class and the raise per plugin error such as `WrongUUIDError(StarletteContextError)`?

Reply:
I agree, it's better to have a specific Exception type, and it might be useful if people raise errors from custom plugin subclasses we don't control.